Respect proxy settings when streaming the download and extraction of the CodeQL bundle (#2624)
github/codeql-action: CHANGELOG.md

## [UNRELEASED]

- We are rolling out a change in December 2024 that will extract the CodeQL bundle directly to the toolcache to improve performance. [#2631](https://github.com/github/codeql-action/pull/2631)

> - We have added support for respecting the proxy settings present in environment variables in a runner when streaming the download and extraction of the CodeQL bundle. [#2624](https://github.com/github/codeql-action/pull/2624)

Reviewer: I'd suggest we phrase this as a bug, since streaming should have respected proxy settings from the get go:

```suggestion
- Fixed an issue where streaming the download and extraction of the CodeQL bundle did not respect proxy settings. [#2624](https://github.com/github/codeql-action/pull/2624)
```

Author: Done.

## [UNRELEASED]

- We are rolling out a change in December 2024 that will extract the CodeQL bundle directly to the toolcache to improve performance. [#2631](https://github.com/github/codeql-action/pull/2631)
- Fixed an issue where streaming the download and extraction of the CodeQL bundle did not respect proxy settings. [#2624](https://github.com/github/codeql-action/pull/2624)

> - Update default CodeQL bundle version to 2.20.0. [#2636](https://github.com/github/codeql-action/pull/2636)

Reviewer 2: Was this line added due to a bad merge?

Author: Hm, I will have a look.

This was reported as a conflict in the GitHub UI, but when I did a `git merge` locally, it had no problem automatically resolving the conflict. Perhaps it did a weird thing because of my `.gitconfig`.

Author: I cannot find precisely what went wrong here, as the merge log is past my terminal character limit. I remember it saying that it resolved it automatically using the `ort` method, but it apparently went wrong there.

This is now fixed in [9e8cd42](https://github.com/github/codeql-action/pull/2624/commits/9e8cd42adc33391cddeaec148d34c6cd8b1d1d65).

Going forward I will be more careful validating these merges went right.
